lib/services/notification-api-service.js

```diff
@@ -70,14 +72,47 @@ function notificationApiServiceFactory(
         });
     };
 
-    notificationApiService.prototype.postBroadcastNotification = function(message) {
-        var self = this;
-
+    NotificationApiService.prototype.postBroadcastNotification = function(message) {
         return eventsProtocol.publishBroadcastNotification(message)
         .then(function () {
             return message;
         });
     };
 
-    return new notificationApiService();
+    NotificationApiService.prototype.postProgressEvent = function(data) {
+        var progressData;
+        return waterline.taskdependencies.find({taskId: data.taskId})
```

Comment:
Copy @yyscamper 's comments at on-taskgraph:

```
use 'findOneinstead offind, so that you can usegraphObjectrather thangraphObject[0]` in following code.
```